src/grammar/cursor.rs

```diff
@@ -0,0 +1,487 @@
+          Whitespace@291..292 " "
+          IdentGroup@292..303
+            Ident@292..303 "departments"
+          Whitespace@303..317 "              "
+          Comment@317..341 "-- repeating return type"
+          Whitespace@341..346 "\n    "
+          WhereClause@346..371
+            Keyword@346..351 "WHERE"
+            Whitespace@351..352 " "
+            Expression@352..371
+              IdentGroup@352..365
+                Ident@352..365 "department_id"
+              Whitespace@365..366 " "
+              ComparisonOp@366..367 "="
+              Whitespace@367..368 " "
+              Integer@368..371 "110"
+          Semicolon@371..372 ";"
+        Whitespace@372..377 "\n \n  "
+      CursorStmt@377..412
+        Keyword@377..383 "CURSOR"
+        Whitespace@383..384 " "
+        IdentGroup@384..386
+          Ident@384..386 "c3"
+        Whitespace@386..387 " "
+        Keyword@387..393 "RETURN"
+        Whitespace@393..394 " "
+        RowtypeClause@394..411
+          IdentGroup@394..403
+            Ident@394..403 "locations"
+          Percentage@403..404 "%"
+          Keyword@404..411 "ROWTYPE"
+        Semicolon@411..412 ";"
+      Whitespace@412..418 "      "
+      Comment@418..431 "-- Declare c3"
+      Whitespace@431..436 "\n \n  "
+      CursorStmt@436..587
+        Keyword@436..442 "CURSOR"
+        Whitespace@442..443 " "
+        IdentGroup@443..445
+          Ident@443..445 "c3"
+        Whitespace@445..446 " "
+        Keyword@446..448 "IS"
+        Whitespace@448..477 "                      ..."
+        Comment@477..490 "-- Define c3,"
+        Whitespace@490..495 "\n    "
+        SelectStmt@495..586
+          Keyword@495..501 "SELECT"
+          Whitespace@501..502 " "
+          Asterisk@502..503 "*"
+          Whitespace@503..504 " "
+          Keyword@504..508 "FROM"
+          Whitespace@508..509 " "
+          IdentGroup@509..518
+            Ident@509..518 "locations"
+          Whitespace@518..534 "                "
+          Comment@534..557 "-- omitting return type"
+          Whitespace@557..562 "\n    "
+          WhereClause@562..585
+            Keyword@562..567 "WHERE"
+            Whitespace@567..568 " "
+            Expression@568..585
+              IdentGroup@568..578
+                Ident@568..578 "country_id"
+              Whitespace@578..579 " "
+              ComparisonOp@579..580 "="
+              Whitespace@580..581 " "
+              QuotedLiteral@581..585 "'JP'"
+          Semicolon@585..586 ";"
+        Whitespace@586..587 "\n"
+    Keyword@587..592 "BEGIN"
+    Whitespace@592..595 "\n  "
+    BlockStatement@595..600
+      Keyword@595..599 "NULL"
+      Semicolon@599..600 ";"
+    Whitespace@600..601 "\n"
+    Keyword@601..604 "END"
+    Semicolon@604..605 ";"
+"#]],
+            vec![],
+        );
+    }
+
+    #[test]
+    fn test_parse_cursor_parameters() {
+        check(
+            parse(
+                "CURSOR c (job VARCHAR2, max_sal NUMBER) IS
```

Comment:
Please also add at least one occurrence of `:= <expression>` or `DEFAULT <expression>` to the test

Reply:
On it